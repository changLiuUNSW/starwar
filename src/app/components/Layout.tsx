import { Container, Navbar, NavbarBrand } from 'reactstrap';
import styled from 'styled-components';

const Main = styled(Container)`
  padding: 1.5rem 0;
`;

interface Props {
  children: React.ReactNode;
}

const Layout = ({ children }: Props) => (
  <div>
    <Navbar color="light" light>
      <NavbarBrand href="/">Star War</NavbarBrand>
    </Navbar>
    <Main tag="main">{children}</Main>
  </div>
);

export default Layout;
