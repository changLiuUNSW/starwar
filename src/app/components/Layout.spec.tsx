import renderer from 'react-test-renderer';
import Layout from './Layout';

it('Layout renders correctly', () => {
  const tree = renderer.create(<Layout>test</Layout>).toJSON();
  expect(tree).toMatchSnapshot();
});
