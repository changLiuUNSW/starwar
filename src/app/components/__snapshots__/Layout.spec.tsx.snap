// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Star War
    </a>
  </nav>,
  <main
    className="sc-bdfBwQ ijcHcs container"
  >
    test
  </main>,
]
`;
